modules/bamcmp/main.nf

```diff
@@ -0,0 +1,48 @@
+def VERSION = '2.2'
+include { initOptions; saveFiles; getSoftwareName; getProcessName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process BAMCMP {
+    tag "$meta.id"
+    label 'process_low'
+    publishDir "${params.outdir}"
+
+
+    conda (params.enable_conda ? "bioconda::bamcmp=2.2" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/bamcmp:2.2--h05f6578_0"
+    } else {
+        container "quay.io/biocontainers/bamcmp:2.2--h05f6578_0"
+    }
+
+    input:
+    tuple val(meta), path(sample), path(contaminant)
+
+    output:
+    tuple val(meta), path("*primary.bam"), emit: bam
+    tuple val(meta), path("*contamination.bam"), emit: contamination_bam
+    path "versions.yml"           , emit: versions
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+
+    """
+    bamcmp \\
+        -s "as" \\
```

Comment:
should this be part of the `$args` or is it mandatory?

Reply:
Also where is `$args` used here?

Reply:
Hi Mahesh, thank you for all your feedback and help with this PR. I hope my container part of code is up to date now. We strongly suggest using the "as" mode but I change it to be optional.